### PR TITLE
Metadata drawer

### LINKED ISSUE #4: Visualise a gallery
Visualise a gallery (collection) or images (items) with thumbnails, image names and image-wise labels.

We will use piximi as the basis for this work, contributing back to the project if we add anything useful.

### PR DESCRIPTION
**Pull request by philipjackson**
# Description

Adds a side-drawer for displaying metadata when a tile is selected (also adds the ability to select tiles). 
Also adds filenames under tiles.

closes #4 

Also, should this be merged into the staging branch? It doesn't look like staging is being used much so far: [https://github.com/gliff-ai/curate/network](https://github.com/gliff-ai/curate/network)

# Dependency changes

N/A

# Testing

N/A

# Documentation

N/A

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] New migrations have been committed
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/ui.tsx [hunk 1]
```diff
@@ -138,45 +164,77 @@ export class UserInterface extends Component<Props, State> {
...
+                    style={{
+                      backgroundColor:
+                        this.state.selected === index && "lightblue",
+                    }}
+                  >
                     <img
                       height={128}
                       src={`data:image/png;base64,${tile.thumbnail}`}
                       alt={tile.name}
+                      onClick={() => {
```
Comment (cooper667) [inline src/ui.tsx:216]: The solution to this is probably wrap the image in an `a` tag?

Comment (philipjackson) [reply src/ui.tsx]: Perhaps that is the answer, but I went with `Button` before I saw this and it turned out to look cooler than I expected.

Comment (cooper667) [reply src/ui.tsx]: yea, that's probably the correct way anyway!

File: src/MetadataDrawer.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      <ListItem style={{ justifyContent: "space-between" }}>
+        <Typography variant="h4">Metadata</Typography>
+        <IconButton onClick={props.handleDrawerClose}>
+          <ChevronRightIcon />
+        </IconButton>
+      </ListItem>
+      <Divider />
+
+      <List>
+        {Object.keys(props.metadata)
```
Comment (SilviaZeta) [inline src/MetadataDrawer.tsx:71]:
```suggestion
      <List dense>
        {Object.keys(props.metadata)
```
To make the metadata fields more compact?

Comment (philipjackson) [reply src/MetadataDrawer.tsx:71]: Do you think they need it? We don't seem to be short on vertical space in the drawer.

File: src/ui.tsx [hunk 3]
```diff
@@ -161,48 +189,89 @@ export class UserInterface extends Component<Props, State> {
...
+                  >
+                    <img
+                      height={128}
+                      src={`data:image/png;base64,${tile.thumbnail}`}
+                      alt={tile.name}
+                    />
+                  </Button>
+                  <Typography style={{ textAlign: "center" }}>
+                    {tile.name.split("/").pop()}
+                  </Typography>
```
[the remark this answers is not in the document]
Comment (philipjackson) [reply src/ui.tsx:259]: Thanks, this is the sort of thing I was looking for when I started working on CURATE. I don't think it does wrapping though; you have to specify the number of columns and the examples don't seem to wrap the tiles if I shrink the browser window. Also I couldn't get it to work, kept getting `TS2604: JSX element type 'ImageListItem' does not have any construct or call signatures.` whenever I try to use `ImageList`, `ImageListItem` or `ImageListItemBar`.

[the remark this answers is not in the document]
Comment (SilviaZeta) [reply src/ui.tsx:259]: It looked good, so i suggested it. I haven't tried it, though (sorry).

**Review by SilviaZeta — APPROVED** [review]

**Review by ChrisBaidoo — APPROVED** [review]

**Pull request merged**